src/distributions/bernoulli.rs

```diff
@@ -67,6 +67,22 @@ impl Bernoulli {
         };
         Bernoulli { p_int }
     }
+
+    /// Construct a new `Bernoulli` with the probability of success of
+    /// `numerator`-in-`denominator`. I.e. `new_ratio(2, 3)` will return
+    /// a `Bernoulli` with a 2-in-3 chance, or about 67%, of returning `true`.
+    ///
+    /// If `numerator == denominator` then the returned `Bernoulli` will always
+    /// return `true`. If `numerator == 0` it will always return `false`.
+    ///
+    /// # Panics
+    ///
+    /// If `denominator` == 0 or `numerator` > `denominator`.
+    ///
+    #[inline]
+    pub fn new_ratio(numerator: u32, denominator: u32) -> Bernoulli {
+        Self::new(numerator as f64 / denominator as f64)
+    }
```

Comment:
If you change this a bit it can improve performance:
```rust
    pub fn new_ratio(numerator: u32, denominator: u32) -> Bernoulli {
        assert!(numerator <= denominator);
        if numerator == denominator {
            return Bernoulli { p_int: ::core::u64::MAX }
        }
        const SCALE: f64 = 2.0 * (1u64 << 63) as f64;
        let p_int = ((numerator as f64 / denominator as f64) * SCALE) as u64;
        Bernoulli { p_int }
    }
```

Before (with `Rng::gen_ratio` changed to use `Bernoulli:new_ratio`):
```
test misc_gen_ratio_const            ... bench:       3,961 ns/iter (+/- 2)
test misc_gen_ratio_var              ... bench:       8,383 ns/iter (+/- 31)
```

After:
```
test misc_gen_ratio_const            ... bench:       3,980 ns/iter (+/- 9)
test misc_gen_ratio_var              ... bench:       7,351 ns/iter (+/- 84)
```

It would be nice if there was a method to calculate the 'cutoff integer' more accurately than converting to floats and back, but I can't think of anything that is not 15× slower.